pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -172,16 +167,19 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*v1.Pod) (*Results, error)
 		s.recordSchedulingResults(ctx, pods, q.List(), errors)
 	}
 	// clear any nil errors so we can know that len(PodErrors) == 0 => all pods scheduled
+	// we can also clear any pod errors for pods that aren't provisionable since
+	// 1. Provisioning only cares about the new nodes we need to provisionable
+	// 2. Deprovisioning only cares about the pods on the candidates we're trying to delete.
 	for k, v := range errors {
-		if v == nil {
+		if v == nil || !pod.IsProvisionable(k) {
```

Comment:
Is there a case where we even handle Provisionable pods? I know that the PendingPods call already filters these out. Are we just handling pods on Deleting nodes here?